### PR TITLE
feature/43: added notifications screen

### PR DESCRIPTION
**Pull request by alena-andr**

### CODE REVIEW

**Comment by Gerasimov94** [comment]
please, add a branch name in the front of your title, for example:
feature/CRC-43: added notifications screen

File: src/navigation/components/roomsNavigation.tsx [hunk 1]
```diff
@@ -20,6 +21,9 @@ export default createStackNavigator({
 			Statistics: {
 				screen: Statistics,
 			},
+			Notifications: {
+				screen: Notifications,
```
Comment (Gerasimov94) [inline src/navigation/components/roomsNavigation.tsx:41]: it's fine by me, but you can write just Notifications, Settings and so on

File: src/screens/Notifications/Notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			}
+		}
+	}
+
+	public render() {
+		return (
+			<ScrollView style={styles.container}>
+				<View style={styles.border} />
+				{
+					arrNotifications.map(notification => this.renderContent(notification))
```
Comment (Gerasimov94) [inline src/screens/Notifications/Notifications.tsx]: arrNotifications.map(this.renderContent)

File: src/screens/Notifications/Notifications.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		model: 'ASI-09HS4/ASO-09HS4',
+		serial: 'ad76aaa0-838f-4f0c-b94e-8c0d1d2e41eb',
+	},
+];
+
+class Notifications extends PureComponent {
+	public renderContent = (notification: any) => {
+		switch (notification.type) {
+			case 'notification': {
+				return <NotificationCard notification={notification} key={notification.id} />;
```
Comment (Gerasimov94) [inline src/screens/Notifications/Notifications.tsx]: it's a good place for React.Lazy, but we have a 16.5 in expo :(

File: src/screens/Notifications/components/NotificationCard/NotificationCard.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {View, Text, Image} from 'react-native';
+
+import generateName from 'sillyname';
```
Comment (Gerasimov94) [inline src/screens/Notifications/components/NotificationCard/NotificationCard.tsx]: i don't like this way for generation, we just may use https://randomuser.me/ . We must remove it when we add a actual data

File: src/screens/Notifications/components/NotificationCard/NotificationCard.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		<View style={styles.container}>
+			<View style={styles.dataWrapper}>
+				<Text style={styles.date}>{date}</Text>
+			</View>
+			<View style={styles.contentWrapper}>
+				<View style={styles.userInfo}>
+					<Image style={styles.image} source={{uri: URI}} />
+					<View style={styles.info}>
+						<Text style={styles.time}>{time}</Text>
+						<Text>
```
Comment (Gerasimov94) [inline src/screens/Notifications/components/NotificationCard/NotificationCard.tsx]: looks like View

File: src/screens/Notifications/components/NotificationCard/NotificationCard.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+						</Text>
+						<Text>
+							<Text style={styles.textBold}>Operation: </Text>
+							<Text style={{color: operation === 'Scheduled' ? '#1ca36e' : '#2582e2'}}>{operation}</Text>
+						</Text>
+					</View>
+				</View>
+				<View style={styles.indicatorWrapper}>
+					<View style={styles.thermometer}>
+						<Thermometer fill='#000000' width={28} height={28} />
```
Comment (Gerasimov94) [inline src/screens/Notifications/components/NotificationCard/NotificationCard.tsx]: move it to colors variables, and check it in all project (variations - #000 #000000)

File: src/screens/Notifications/notificationsStyles.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+	container: {
+		flex: 1,
+	},
+	border: {
+		position: 'absolute',
+		top: 0,
+		left: 0,
+		width: 74,
+		height: '100%',
+		backgroundColor: '#f7f7f7',
```
Comment (Gerasimov94) [inline src/screens/Notifications/notificationsStyles.ts]: to variable, same at  notificationCardStyles.ts,  servicesNotificationCardStyles.ts

File: src/screens/Notifications/components/ServicesNotificationCard/servicesNotificationCardStyles.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+		display: 'flex',
+		alignItems: 'center',
+		width: 74,
+		backgroundColor: 'transparent',
+	},
+	date: {
+		fontSize: 14,
+		fontWeight: 'bold',
+		paddingTop: 36,
+		color: '#000000',
```
Comment (Gerasimov94) [inline src/screens/Notifications/components/ServicesNotificationCard/servicesNotificationCardStyles.ts]: same

**Review by Gerasimov94 — CHANGES_REQUESTED** [review]

File: src/screens/Notifications/components/NotificationCard/NotificationCard.tsx (hunk 4 above)
Comment (CodeAnnihilator) [reply src/screens/Notifications/components/NotificationCard/NotificationCard.tsx]: they are the same, and we will get rid of this dummy images soon. We will have fixtures on a server side to generate images there. Let's keep 'sillyname' module instead, at leas for now.

File: src/screens/Notifications/components/NotificationCard/NotificationCard.tsx (hunk 6 above)
Comment (CodeAnnihilator) [reply src/screens/Notifications/components/NotificationCard/NotificationCard.tsx]: good catch. We will have a separate task for that right away

**Review by Gerasimov94 — APPROVED** [review]

**Review by CodeAnnihilator — CHANGES_REQUESTED** [review]
remove CRC

File: src/screens/Notifications/Notifications.tsx (hunk 2 above)
Comment (alena-andr) [reply src/screens/Notifications/Notifications.tsx]: done

File: src/screens/Notifications/components/NotificationCard/NotificationCard.tsx (hunk 5 above)
Comment (alena-andr) [reply src/screens/Notifications/components/NotificationCard/NotificationCard.tsx]: done

File: src/common/seed/entities.ts [hunk 9]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (CodeAnnihilator) [inline src/common/seed/entities.ts]: fix this

File: src/screens/Notifications/containers/NotificationsContainer.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import {connect} from 'react-redux';
+
+import Notifications from '../Notifications';
+import {sortUniqDate, getElsByDay} from '../selectors/notificationsSelectors';
+
+const mapStateToProps = (state: any) => ({
+	notifications: getElsByDay(state),
+	dates: sortUniqDate(state),
```
Comment (CodeAnnihilator) [inline src/screens/Notifications/containers/NotificationsContainer.ts]: sortByUniqueDate

**Review by CodeAnnihilator — CHANGES_REQUESTED** [review]

File: src/common/seed/entities.ts (hunk 9 above)
Comment (alena-andr) [reply src/common/seed/entities.ts]: done

File: src/screens/Notifications/containers/NotificationsContainer.ts (hunk 10 above)
Comment (alena-andr) [reply src/screens/Notifications/containers/NotificationsContainer.ts]: done

**Review by CodeAnnihilator — APPROVED** [review]

**Pull request merged**